backend/server-connection/src/main/java/org/sonarsource/sonarlint/core/serverconnection/AnalyzerConfigurationStorage.java

```diff
@@ -131,6 +136,12 @@ private static Sonarlint.RuleSet.ActiveRule adapt(ServerActiveRule rule) {
       .setSeverity(rule.getSeverity().name())
       .setTemplateKey(rule.getTemplateKey())
       .putAllParams(rule.getParams())
+      .addAllOverriddenImpacts(rule.getImpacts() == null ? List.of() : rule.getImpacts().stream()
```

Comment:
Since you made the same check somewhere else, ServerActiveRule should never be empty I think? So you could remove the check and directly use the value